www/activities/foods-meals-recipes/js/food-editor.js

```diff
@@ -17,6 +17,35 @@
   along with app.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+
+
+function evalInpEquation(strIn)
+{
+  let ret = 0;
+  let strInTmp = strIn.trim();
+
+  if (strInTmp.length == 0 || strInTmp == "+" || strInTmp == "-")
+    return NaN;
+
+  let cleanStr = strInTmp.replaceAll("+", " + ").replaceAll("-", " - ").replaceAll("  ", " ");
+  let tokens = cleanStr.split(" ");
```

Comment:
It would be better and safer to simply split the string at any whitespace characters with a regular expression: `cleanStr.split(/\s+/)`. This way, you also don't need to worry about potential double or triple spaces.